test/common/common/assert_test.cc

```diff
@@ -42,4 +42,29 @@ TEST(AssertDeathTest, VariousLogs) {
   EXPECT_EQ(expected_counted_failures, assert_fail_count);
 }
 
+TEST(EnvoyBugDeathTest, VariousLogs) {
+  int expected_counted_failures;
+  int envoy_bug_fail_count = 0;
+  // ENVOY_BUG actions only occur on power of two counts.
+  auto envoy_bug_action_registration =
+      Assert::setEnvoyBugFailureRecordAction([&]() { envoy_bug_fail_count++; });
+
+#ifndef NDEBUG
+  EXPECT_DEATH({ ENVOY_BUG(false); }, ".*envoy bug failure: 0.*");
+  EXPECT_DEATH({ ENVOY_BUG(false, ""); }, ".*envoy bug failure: 0.*");
+  EXPECT_DEATH({ ENVOY_BUG(false, "With some logs"); },
+               ".*envoy bug failure: 0. Details: With some logs.*");
```

Comment:
informational only, no action needed:

There's something interesting about ENVOY_BUG in debug modes that we may want to comment about here:
Exponential backoff doesn't seem to be triggering for some reason.  Fortunately, that's a good thing: we want ENVOY_BUG to act like an ASSERT always in debug modes so we can more easily detect when it happens and ensure that our test cases do not become order dependent.  I think this difference in behavior is related to the use of EXPECT_DEATH which does a fork; count is at 0 when each of the ENVOY_BUGs above execute.

Possible suggestion: it would be good to have macros that make it easier to test for code triggering ENVOY_BUG.  I guess we can use EXPECT_DEBUG_DEATH until a more specialized macro exists.